### DISCUSSION TITLE
Clear input event queues??

### DISCUSSION THREAD

**Discussion by kkelly3137** [gpiozero/gpiozero#1065] [Q&A]
Does gpiozero have any mechanisms to clear the interrupt queue(s) of a GPIO input?  Not for switch denounce purposes but if the user "just keeps pressing the button" it can cause undesired results of the event triggering multiple times.

My application is doorbell that is connected to the Pi the calls a phone number or plays a tone file based on a schedule.  Using the mybutton.when_held timer (BTW  -- thanks sooo much!!! It **SOLVED** a bunch of RF noise random trigger issues) but it  doesn't stop the user from pressing the button multiple times within the when_held timer .  

The events that can be triggered can take 6 - 15 seconds to resolve (play a tone or dial a phone number, wait for answer) so clearing the input queue after the event would be desirable.

Again great code! Thx

-Kevin-

**Comment by lurch** [comment]
It's a bit of a hacky solution (because it uses global variables), but perhaps an approach like this pseudo-code might work?

```
button_pressed = False
sound_playing = False

def button_callback():
    global button_pressed
    global sound_playing
    if sound_playing:
        return
    # button_pressed can only be set to True, when sound_playing is False
    button_pressed = True

# set up GPIOs, callbacks, etc...

while True:
    if button_pressed:
        sound_playing = True
        button_pressed = False
        play_sound() # this can take several seconds
        sound_playing = False
    sleep(1)
````

This way, the `button_callback` always completes quickly (i.e. the button-press gets ignored if the sound is still playing), and so you don't get a backlog of queued input events?

**Reply by kkelly3137** [reply]
I already do something similar inside a "playWav()" function where it may just play a 3 second sound or make a phone call (15 seconds) and I return() out of that function if its called again and is still playing, calling etc.  Didn't think of doing the same in the button function.  Good idea, thanks.  

My other thought was to turn the GPIO pin off and on in hardware, but it just feels wrong doing all those highlevel hardware calls.
